BSP/Simulator_V2/Src/Simulator.c

```diff
@@ -0,0 +1,227 @@
+/* Copyright (c) 2022 UT Longhorn Racing Solar */
+/**
+ * This file creates an interface for the simulator to create a trace for running tests
+ */
+
+#include "Simulator.h"
+#include <time.h> // for creating unique logfile names
+// for open() syscall
+#include <sys/types.h>  // for stat() and struct stat as well
+#include <sys/stat.h>
+#include <fcntl.h>
+// for write() and read()
+#include <unistd.h>
+// for sprintf
+#include <stdio.h>
+// for signal handler
+#include <signal.h>
+// for boolean
+#include <stdbool.h>
+#include "cJSON.h" // for json parser
+#include <stdlib.h> // for calloc/free
+
+// file descriptor of simulator log file
+int simulatorLog;
+
+// linked list of simulator states
+static simulator_state *states = NULL;
+
+// timing information to handle the state transitions
+// note that this only has a granularity of 1 second, so it's not super precise
+static time_t startTime;
+
+// It knows what the input file is because the Makefile should make a #define for the file path called SIMULATOR_JSON_PATH
+#ifndef SIMULATOR_JSON_PATH
+#define JSON_PATH "Error: make sure you define the simulator JSON path in the Makefile!"
+#else
+// #define QUOTE_MARK "
+#define STR(X) # X
+#define STRINGIFY(X) STR(X)
+#define JSON_PATH STRINGIFY(SIMULATOR_JSON_PATH) 
+#endif
+
+// log something to the simualtor's log file
+void Simulator_log(char *str) {
```

Comment:
Should this have parameters similar to python logging, where you log errors, info, or warnings? I feel it would be easier to tell if something broke by just checking for an error instead of making sure when people write log data, they have to write `Hard Fault: ...`

Reply:
Yeah, I can change it to something like that